src/sql/workbench/services/notebook/browser/models/modelInterfaces.ts

```diff
@@ -527,7 +527,7 @@ export interface ICellModel {
 	isEditMode: boolean;
 	showPreview: boolean;
 	showMarkdown: boolean;
-	defaultToWYSIWYG: boolean;
+	defaultTextEditMode: string;
```

Comment:
Why does the model need to store this? Shouldn't this be a one-time thing we check when opening a Notebook (thus you could just put all the logic in the markdown toolbar)?

Reply:
The value gets checked when opening the cell for edit (isEditMode), so we use it in the CellModel class in addition to the markdown toolbar.

Reply:
Fair enough - but same thing still applies. Any reason you wouldn't want to just check the setting each time?

Reply:
The setting probably won't change that often, so caching it as a string field keeps us from having to query the Configuration Service every time, or potentially having to add a reference to the Configuration Service in classes where we didn't need it before.

Reply:
Querying the configuration is cheap so that isn't something that we should generally be concerned about.

I don't really like the idea of attaching a property like this to the model when it isn't something that's actually part of the model. Why would anyone else ever need to access this?

It's not really a problem per-se, but it's just an additional thing I don't need and would be simpler without it there.